Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8841,6 +8841,73 @@ def shorten_string_for_printing(source_string, max_length=64):
         return ret_value
 
 
+class PollResult:
+    def __init__(self, response, continue_to_poll=False, args_for_next_run=None):
+        """
+        Constructor for PollResult
+
+        :type response: ``Any``
+        :param response:  The response of the command in the event of success,
+        or in case of failure but Polling is false
+
+        :type continue_to_poll: ``Union[bool, Callable]``
+        :param continue_to_poll: an iterable of relevant keys list from the json. Notice we save it as a set in the class
+
+        :type args_for_next_run: ``Dict``
+        :param args_for_next_run: The arguments to use in the next iteration. Will use the input args in case of None
+
+        """
+        self.response = response
+        self.continue_to_poll = continue_to_poll
+        self.args_for_next_run = args_for_next_run
+
+
+def poll(name, interval=30, timeout=600, poll_message='Fetching Results:', polling_arg_name="Polling"):
+    """
+    To use on a function that should rerun itself
+    Commands that use this decorator must have a Polling argument, polling: true in yaml,
+    and a hidden polled_once argument.
+    Commands that use this decorator should return a PollResult.
+    ----------
+    name : str
+        The name of the command
+    interval : int
+        How many seconds until the next run
+    timeout : int
+        How long
+    poll_message : str
+    The message to display in the war room while polling
+    Raises
+    ------
+    DemistoException
+        If the server version doesn't support Scheduled Commands (< 6.2.0)
+    """
+
+    def dec(func):
+        def inner(client, args):
+            if args.get(polling_arg_name):
+                ScheduledCommand.raise_error_if_not_supported()
+                poll_result = func(client, args)
+
+                should_poll = poll_result.continue_to_poll if isinstance(poll_result.continue_to_poll, bool) \
+                    else poll_result.continue_to_poll()
+                if not should_poll:
+                    return poll_result.response
+                poll_args = poll_result.args_for_next_run or args
+                readable_output = poll_message if not args.get('polled_once') else None
+                poll_args['polled_once'] = True
+                return CommandResults(readable_output=readable_output,
+                                      scheduled_command=ScheduledCommand(command=name, next_run_in_seconds=interval,
```

Comment:
I think it'll be a better approach for `PollResult` to have the `CommandResults` as part of its properties rather than the `response`. It'll allow more flexibility when creating the `CommandResults` to return. Note that `outputs` (and any other `CommandResults` properties) is an optional part of a valid `PollingResult` - i.e. if we have partial data in the `PollingResult` and we'd like to update the `ContextData` with it before returning the full 
 result.

Reply:
@DeanArbel I added partial_response 
We need to differentiate between a response that will be given with the intent that the command will be polled again, and one where it wont (Polling = false).

Reply:
Ok 👍